novawallet/Common/Services/DelegatedAccounts/SyncService/DelegatedAccountSyncBarrier.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+import Operation_iOS
+
+protocol DelegatedAccountSyncBarrierProtocol {
+    func filter(_ changes: SyncChanges<ManagedMetaAccountModel>) -> SyncChanges<ManagedMetaAccountModel>
+}
+
+final class DelegatedAccountSyncBarrier {
+    private let mutex = NSLock()
+    private let logger: LoggerProtocol
+
+    private var registeredDelegations: Set<MetaAccountDelegationId> = []
+
+    init(logger: LoggerProtocol = Logger.shared) {
+        self.logger = logger
+    }
+}
+
+// MARK: - DelegatedAccountSyncBarrierProtocol
+
+extension DelegatedAccountSyncBarrier: DelegatedAccountSyncBarrierProtocol {
+    func filter(_ changes: SyncChanges<ManagedMetaAccountModel>) -> SyncChanges<ManagedMetaAccountModel> {
+        mutex.lock()
+
+        defer {
+            mutex.unlock()
+        }
+
+        let newOrUpdatedItems = changes.newOrUpdatedItems.filter { delegatedMetaAccount in
+            guard let delegationId = delegatedMetaAccount.info.delegationId() else { return false }
```

Comment:
I think we are skipping here chain information and type here. For example, i imagine we treat the same 2 proxies in different chains or proxy vs multisig